### PR TITLE
Save exchange rate hints to disk to use across all accounts

### PR DESCRIPTION
**Pull request by peachbits**

### CODE REVIEW

File: src/core/root-reducer.js [hunk 1]
```diff
@@ -85,6 +87,15 @@ export const reducer: BuiltReducer<RootState, RootAction> = buildReducer({
     return state
   },
 
+  rateHintCache(state = DEFAULT_RATE_HINTS, action): EdgeRateHint[] {
```
Comment (peachbits) [inline src/core/root-reducer.js:90]: empty array

File: src/core/exchange/exchange-pixie.js [hunk 2]
```diff
@@ -47,6 +50,29 @@ export const exchange: TamePixie<RootProps> = filterPixie(
...
+            'rateHintCache.txt',
+            JSON.stringify(oldRateHintCache.concat(newRateHints))
+          )
+          input.props.dispatch({
+            type: 'UPDATE_RATE_HINT_CACHE',
+            payload: { rateHintCache: oldRateHintCache.concat(newRateHints) }
+          })
+          input.props.log.warn('Update rateHintCache.txt success')
+        } catch (error) {
+          input.props.log.error('Update rateHintCache.txt error', error)
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/core/exchange/exchange-pixie.js]: onError isn't available when doing the initial read but it was added to the write in gatherHints

File: src/core/root.js [hunk 3]
```diff
@@ -71,6 +71,25 @@ export async function makeContext(
...
+    rateHintCache = JSON.parse(await io.disklet.getText('rateHintCache.txt'))
+    log.warn('Read rateHintCache.txt success')
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      // Failure is ok if file doesn't exist
+      try {
+        await io.disklet.setText('rateHintCache.txt', JSON.stringify([]))
+        log.warn('Create rateHintCache.txt success')
+      } catch (error) {
+        log.error('Create rateHintCache.txt failure', error.message)
```
Comment (paullinator) [inline src/core/root.js:86]: This is a catastrophic error condition, throw the error

File: src/core/root.js [hunk 4]
```diff
@@ -71,6 +71,25 @@ export async function makeContext(
...
+      }
+    } else {
+      log.error('Read rateHintCache.txt error', error.message)
```
Comment (paullinator) [inline src/core/root.js:89]: This is a catastrophic error condition, throw the error

Comment (paullinator) [reply src/core/root.js:89]: or call `input.props.onError()`

**Review by paullinator — CHANGES_REQUESTED** [review]

**Review by paullinator — APPROVED** [review]

**Pull request merged**